src/tiledb/cloud/vector_search/file_ingestion.py

```diff
@@ -0,0 +1,200 @@
+    :param workers: If `embeddings_generation_mode=BATCH` this is the number of
+        distributed workers to be used.
+    :param worker_resources: If `embeddings_generation_mode=BATCH` this can be used
+        to specify the worker resources.
+    :param worker_image: If `embeddings_generation_mode=BATCH` this can be used
+        to specify the worker Docker image.
+    :param extra_worker_modules: If `embeddings_generation_mode=BATCH` this can be
+        used to install extra pip package to the image.
+    :param driver_resources: If `embeddings_generation_driver_mode=BATCH` this can
+        be used to specify the driver resources.
+    :param driver_image: If `embeddings_generation_driver_mode=BATCH` this can be
+        used to specify the driver Docker image.
+    :param extra_driver_modules: If `embeddings_generation_driver_mode=BATCH` this
+        can be used to install extra pip package to the image.
+    :param max_tasks_per_stage: Number of maximum udf tasks per computation stage.
+    :param embeddings_generation_mode: TaskGraph execution mode for embeddings
+        generation.
+    :param embeddings_generation_driver_mode: TaskGraph execution mode for the
+        ingestion driver.
+    :param vector_indexing_mode: TaskGraph execution mode for the vector indexing.
+    :param index_update_kwargs: Extra arguments to pass to the index update job.
+    """
+    import tiledb
+    import tiledb.vector_search.embeddings as embeddings_module
+    from tiledb.vector_search.object_api import object_index
+    from tiledb.vector_search.object_readers import DirectoryTextReader
+
+    if environment_variables is None:
+        environment_variables = {}
+    if index_creation_kwargs is None:
+        index_creation_kwargs = {}
+    if index_update_kwargs is None:
+        index_update_kwargs = {"files_per_partition": 100}
+    if embedding_kwargs is None:
+        embedding_kwargs = {
+            "dimensions": 1536,
+            "embedding_class": "OpenAIEmbeddings",
+            "embedding_kwargs": {
+                "model": "text-embedding-ada-002",
+            },
+        }
+    if text_splitter_kwargs is None:
+        text_splitter_kwargs = {
+            "chunk_size": 500,
+            "chunk_overlap": 50,
+        }
+    if driver_resources is None:
+        driver_resources = {"cpu": "2", "memory": "8Gi"}
+
+    reader = DirectoryTextReader(
+        search_uri=file_dir_uri,
+        include=file_name if file_name is not None else include,
+        exclude=exclude,
+        suffixes=suffixes,
+        max_files=max_files,
+        text_splitter=text_splitter,
+        text_splitter_kwargs=text_splitter_kwargs,
+    )
+
+    if openai_key is not None:
+        environment_variables["OPENAI_API_KEY"] = openai_key
+    embedding_class_ = getattr(embeddings_module, embedding_class)
+    embedding = embedding_class_(**embedding_kwargs)
+
+    with tiledb.scope_ctx(config):
```

Comment:
This should be done as the first stage in the graph, not local to the caller.

Reply:
I am not sure I understand this, `ingest_files_udf` is running within the taskgraph

Reply:
I created an alternative version of the PR that uses an extra taskgraph and creates the dataset as a first node in the taskgraph  https://github.com/TileDB-Inc/TileDB-Cloud-Py/pull/547 Is this what you are expecting the ingestion structure to look like?

Reply:
@Shelnutt2 is the alternative version matching your expectation? Let me know if you still have concerns

Reply:
Applied the alternative taskgraph structure in this PR.

Reply:
@Shelnutt2 this PR needs your approval to move forward. Let me know if you need any more changes.